File: Kids_With_The_Greatest_Number_Of_Candies/cpp.d/most_candies.cpp
#include <iostream>
#include <string>
#include <vector>
#include <stdio.h>
using namespace std;


class Solution{
  public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies);
};


vector<bool> Solution::kidsWithCandies(vector<int>& candies, int extraCandies){
  vector<bool> most_candies = {};
  int largest_num = 0;
  for (auto x: candies){
    if (x > largest_num)
      largest_num = x;
  }
  for (auto x: candies){
    if (x + extraCandies >= largest_num){
      most_candies.push_back(true);
    } else {
      most_candies.push_back(false);
    }
  }
  return most_candies;
}


int main(){
  vector<int> candies = {4,2,1,1,2};
  Solution sol;
  vector<bool> hasMost = sol.kidsWithCandies(candies, 1);
  for (auto x: hasMost)
    cout << x << " ";
  cout << endl;
  return 0;
}

File: Kids_With_The_Greatest_Number_Of_Candies/py.d/most_candies.py
from typing import List

class Solution:
  def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
    return [(num_candies + extraCandies >= max(candies)) for num_candies in candies]

if __name__=="__main__":
  sol = Solution()
  candies = [2,3,5,1,3]
  extra_candies = 3
  print(sol.kidsWithCandies(candies, extra_candies)) # => [true,true,true,false,true]
